### PR TITLE
SCons: Misc improvements

### PR DESCRIPTION
**Pull request by ghost**
* Delete old gcc 7 check
* Use f-strings where it is easy
* Use AddVariables instead of Add for collections of options

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]
Looks good to me.

File: SConstruct [hunk 1]
```diff
@@ -854,8 +852,6 @@ else:  # GCC, Clang
 
     if methods.using_gcc(env):
         common_warnings += ["-Wshadow", "-Wno-misleading-indentation"]
-        if cc_version_major == 7:  # Bogus warning fixed in 8+.
-            common_warnings += ["-Wno-strict-overflow"]
```
Comment (adamscott) [inline SConstruct:858]: Do we still support gcc 7?

And is there a check beforehand to let people using gcc 7 know that compiling Godot with it isn't supported?

Comment (dustdfg) [reply SConstruct:858]: We require gcc 9 at minimum

https://github.com/godotengine/godot/blob/9e6098432aac35bae42c9089a29ba2a80320d823/SConstruct#L650-L657

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**